Rewrite the identity mapping explanation and add Key Features section to README (#82)
Azure-Samples/communication-services-messaging-teams-app: README.md

This is a sample application to show how we can build a custom [Teams App](https://docs.microsoft.com/microsoftteams/platform/overview#build-apps-with-microsoft-teams-platform) that can interface with an Azure Communication Services instance, enabling the two systems to work together while keeping their backend environments and identity configurations separate.

This sample includes three standalone applications: CustomerApp, BusinessApp, and Server. Together, they provide a comprehensive customer support experience. The overall view of the system is shown below:
<img src="./Assets/architecture-diagram.png" alt="Architecture Diagram" width="1024">

1. An Azure Communication Services instance that enables the chat experience.

> 2. **CustomerApp**: A web application used by customers to interact with agents. Users of this CustomerApp will be assigned an Azure Communication Services identities.

Reviewer: ```suggestion
2. **CustomerApp**: A web application used by customers to interact with agents. Users of this CustomerApp will be assigned Azure Communication Services identities.
```

3. **BusinessApp**: A web application hosted within a custom Teams application and deployed to Teams through an iframe inside the Teams client. Agents utilize this app within their Teams client.
   - This app leverages Teams’ Single Sign-On (SSO) to retrieve the Teams user ID.

>    - The Azure Communication Services instance is not directly connected to the Teams environment. [Identity mapping](https://learn.microsoft.com/azure/communication-services/concepts/identity-model#user-identity--mapping) is required to connect the BusinessApp users who have the Teams user ID, to the CustomerApp users who have the Azure Communication Services identity.

Reviewer: ```suggestion
   - The Azure Communication Services instance is not directly connected to the Teams environment. [Identity mapping](https://learn.microsoft.com/azure/communication-services/concepts/identity-model#user-identity--mapping) is required to connect the BusinessApp users who have a Teams user ID, to the CustomerApp users' Azure Communication Services identities.
```

>    - For this sample app, an Azure Communication Services identity needs to be manually created for each BusinessApp user, then store the identity mapping in the environment variables ([steps are shown below in the AgentUsers section](https://github.com/Azure-Samples/communication-services-messaging-teams-app?tab=readme-ov-file#before-running-the-sample-for-the-first-time)).

Reviewer: ```suggestion
   - For this sample app, an Azure Communication Services identity needs to be manually created for each BusinessApp user, and the identity mapping stored in environment variables ([steps are shown below in the AgentUsers section](https://github.com/Azure-Samples/communication-services-messaging-teams-app?tab=readme-ov-file#before-running-the-sample-for-the-first-time)).
```

>    - Once the linkage is set, the BusinessApp will use the Teams user ID to map to the Azure Communication Services ID, then use that ID to retrieve threads and communicate with CustomerApp users.

Reviewer: ```suggestion
   - Once the linkage is set, the BusinessApp will use the Teams user ID to retrieve the Azure Communication Services ID, and use that ID to retrieve threads and communicate with CustomerApp users.
```
